### PR TITLE
Various infra setup improvements

### PR DESCRIPTION
**Pull request by AronNovak**

### CODE REVIEW

File: web/sites/default/settings.pantheon.php [hunk 1]
```diff
@@ -69,3 +72,26 @@
       break;
   }
 }
+
+if (!empty($pantheon_env) && !empty($_ENV['CACHE_HOST'])) {
+  // Include the Redis services.yml file. Adjust the path if you installed to a contrib or other subdirectory.
```
Comment (AronNovak) [inline web/sites/default/settings.pantheon.php:77]:
```suggestion
```

File: web/sites/default/settings.pantheon.php [hunk 2]
```diff
@@ -40,6 +40,9 @@
     }
     $_ENV['es_env'] = $env;
 
+    // The port number is MANDATORY, even if it's the default one.
+    // Elastic.co these days put instances on default port, include :443
+    // nevertheless at the end of the URL.
```
Comment (amitaibu) [inline web/sites/default/settings.pantheon.php:45]: Let's change below example then to be `:443`

**Review by amitaibu — APPROVED** [review]

File: web/sites/default/settings.pantheon.php [hunk 3]
```diff
@@ -40,6 +40,9 @@
...
     $config['elasticsearch_connector.cluster.server']['url'] = 'https://REPLACE-WITH-REAL-URL.us-central1.gcp.cloud.es.io:9243';
```
Comment (AronNovak) [inline web/sites/default/settings.pantheon.php:46]:
```suggestion
    $config['elasticsearch_connector.cluster.server']['url'] = 'https://REPLACE-WITH-REAL-URL.us-central1.gcp.cloud.es.io:443';
```

File: web/sites/default/settings.pantheon.php [hunk 4]
```diff
@@ -69,3 +72,25 @@
...
+
+if (!empty($pantheon_env) && !empty($_ENV['CACHE_HOST'])) {
+  $settings['container_yamls'][] = 'modules/contrib/redis/redis.services.yml';
+  $settings['container_yamls'][] = 'modules/contrib/redis/example.services.yml';
+
+  // phpredis is built into the Pantheon application container.
+  $settings['redis.connection']['interface'] = 'PhpRedis';
+  // These are dynamic variables handled by Pantheon.
+  $settings['redis.connection']['host']      = $_ENV['CACHE_HOST'];
+  $settings['redis.connection']['port']      = $_ENV['CACHE_PORT'];
```
Comment (amitaibu) [inline web/sites/default/settings.pantheon.php:84]:
```suggestion
  $settings['redis.connection']['host'] = $_ENV['CACHE_HOST'];
  $settings['redis.connection']['port'] = $_ENV['CACHE_PORT'];
```

File: web/sites/default/settings.pantheon.php [hunk 5]
```diff
@@ -69,3 +72,25 @@
...
+if (!empty($pantheon_env) && !empty($_ENV['CACHE_HOST'])) {
+  $settings['container_yamls'][] = 'modules/contrib/redis/redis.services.yml';
+  $settings['container_yamls'][] = 'modules/contrib/redis/example.services.yml';
+
+  // phpredis is built into the Pantheon application container.
+  $settings['redis.connection']['interface'] = 'PhpRedis';
+  // These are dynamic variables handled by Pantheon.
+  $settings['redis.connection']['host'] = $_ENV['CACHE_HOST'];
+  $settings['redis.connection']['port'] = $_ENV['CACHE_PORT'];
+  $settings['redis.connection']['password']  = $_ENV['CACHE_PASSWORD'];
```
Comment (AronNovak) [inline web/sites/default/settings.pantheon.php:85]:
```suggestion
  $settings['redis.connection']['password'] = $_ENV['CACHE_PASSWORD'];
```

**Pull request merged**